### PR TITLE
Added Polymorph adapter

### PR DESCRIPTION
**Pull request by kuldeepk**
First commit for Polymorph Adapter

### CODE REVIEW

**Comment by jaiminpanchal27** [comment]
@kuldeepk Did you get a chance to look at changes requested ?

File: modules/polymorphBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      var crid = '';
+      if (typeof serverResponse.ad !== 'undefined') {
+        crid = serverResponse.crid;
+        theHTML = serverResponse.ad.html;
+        width = serverResponse.ad.width || width
+        height = serverResponse.ad.height || height
+      } else {
+        crid = serverResponse.ads[0].crid;
+        theHTML = serverResponse.html;
+      }
```
[the remark this answers is not in the document]
Comment (kuldeepk) [reply modules/polymorphBidAdapter.js]: @jaiminpanchal27 I don't follow. I don't see mediaType as a part of bidResponse in here - http://prebid.org/dev-docs/bidder-adaptor.html#interpreting-the-response or in your adapter example. What am I missing?

**Comment by kuldeepk** [comment]
@jaiminpanchal27 thanks for the reminder. Can you tell me why is it tagged as `needs docs`? We've added the docs here - https://github.com/picatcha/Prebid.js/blob/master/modules/polymorphBidAdapter.md

File: modules/polymorphBidAdapter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaiminpanchal27) [reply modules/polymorphBidAdapter.js]: @kuldeepk  Its not explicitly mentioned on the page but if you scroll down a little you will find `Step 3: Respond with a VAST URL or raw VAST XML` you will see a code block. We will update the docs.

Prebid supports multi format, hence its important for us to know the type of bid response to add bid properly to auction.

**Comment by jaiminpanchal27** [comment]
@kuldeepk  Please submit a PR to the [docs repo](https://github.com/prebid/prebid.github.io) to add a file for your adapter to the [bidders directory](https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders) so your adapter's params will appear on the [bidders page](http://prebid.org/dev-docs/bidders.html). Thank you for contributing

**Comment by kuldeepk** [comment]
@jaiminpanchal27 Thanks. I've submitted the PR for documentation - https://github.com/prebid/prebid.github.io/pull/856. Let me know if there are any other steps. Thanks for your support.

**Comment by jaiminpanchal27** [comment]
@kuldeepk Can you address this as well https://github.com/prebid/Prebid.js/pull/2526#discussion_r196215281

File: modules/polymorphBidAdapter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kuldeepk) [reply modules/polymorphBidAdapter.js]: I updated here - https://github.com/prebid/Prebid.js/pull/2526/files#diff-9f4e7fe5997e16525d3ff546f7989200R96. Let me know if that looks okay.

**Comment by mkendall07** [comment]
This pull request **introduces 2 alerts** when merging 9744569c8991d90f9f5391f25a9e0a995f165994 into 22497787bc57281f93aa2089509c240d1e371dd4 - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-dce5c66cb319b4fad9e3154a54ea74b83be5f9ee)

**new alerts:**

* 2 for Unused variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by mkendall07** [comment]
This pull request **introduces 2 alerts** when merging d8d3658a580f3b8febcf2b95bbd9613774808a66 into 22497787bc57281f93aa2089509c240d1e371dd4 - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-8adeb3f51d8805bafae1ad6dd18e8291d03f5898)

**new alerts:**

* 2 for Unused variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by kuldeepk** [comment]
@jaiminpanchal27 Thanks for the reviewing and approving the changes. Are there any next steps we need to do merge this in the master branch and getting released to live release?

**Pull request merged**